m2install.sh

```diff
@@ -640,6 +666,102 @@ function configure_files()
     patchDumps
 }
 
+function add_remote()
+{
+    updateEnvFileRemote
+    patchRemote
+}
+
+function updateEnvFileRemote()
+{
+    local deployConfigurator=$(cat << EOF
+<?php
+
+\$dbName = '${REMOTE_DB}';
+\$dbPassword = '${REMOTE_DB_PASSWORD}';
+\$localPort = '${LOCAL_PORT}';
+
+EOF
+);
+    deployConfigurator+=$(cat << 'EOF'
+
+function updateDbConnection($envConfig, $connectionDetails)
+{
+    unset($envConfig['db']['slave_connection']);
+
+    foreach ($envConfig['db'] as $key => $connections) {
+        if ($key != 'connection') {
+            continue;
+        }
+        foreach ($connections as $connectionName => $connectionParams) {
+            $envConfig['db'][$key][$connectionName] = $connectionDetails;
+        }
+    }
+
+    return $envConfig;
+}
+
+$envConfig = require 'app/etc/env.php';
+$db = explode("_", $dbName);
+$envConfig = updateDbConnection($envConfig, array(
+    'host' => "127.0.0.1:$localPort",
+    'dbname' => $dbName,
+    'username' => $db[0],
+    'password' => "$dbPassword",
+    'model' => 'mysql4',
+    'engine' => 'innodb',
+    'initStatements' => 'SET NAMES utf8;',
+    'active' => '1'
+));
+
+echo "<?php\nreturn " . var_export($envConfig, true) . "\n;";
+EOF
+);
+
+ echo "$deployConfigurator" | ${BIN_PHP} > app/etc/env.php.generated
+ mv app/etc/env.php.generated app/etc/env.php
+}
+
+function patchRemote()
+{
+    local patchBody+="
++
++//patched by m2install.
++\$_ENV['CONFIG__DEFAULT__WEB__UNSECURE__BASE_URL'] = '${HTTP_HOST}';
++\$_ENV['CONFIG__DEFAULT__WEB__SECURE__BASE_URL'] = '${HTTP_HOST}';
++\$dbHostPort = '${REMOTE_DB_HOST}';
++\$sshHost = '${REMOTE_HOST}';
++\$sshKey = '${REMOTE_KEY}' ? '-i ${REMOTE_KEY}' : '';
++\$localPort = '${LOCAL_PORT}';
+";
+    patchBody+=$(cat << 'EOF'
++$command = "ssh $sshKey -o StrictHostKeyChecking=no -fN -L $localPort:$dbHostPort $sshHost";
```

Comment:
do not connecting without forcing IPv4

ssh $sshKey -o StrictHostKeyChecking=no -4fN -L $localPort:$dbHostPort $sshHost